azkaban-common/src/main/java/azkaban/imagemgmt/permission/PermissionManagerImpl.java

```diff
@@ -42,25 +44,24 @@ public class PermissionManagerImpl implements PermissionManager {
   private static final Logger log = LoggerFactory.getLogger(PermissionManagerImpl.class);
 
   private final ImageTypeDao imageTypeDao;
+  private final UserManager userManager;
 
   @Inject
-  public PermissionManagerImpl(final ImageTypeDao imageTypeDao) {
+  public PermissionManagerImpl(final ImageTypeDao imageTypeDao, final UserManager userManager) {
     this.imageTypeDao = imageTypeDao;
+    this.userManager = userManager;
   }
 
   /**
    * Checks the permission based on user manager, image type name, user id and Permission type.
    *
-   * @param userManager
    * @param imageTypeName
    * @param userId
    * @param type
    * @return boolean
    */
   @Override
-  public boolean hasPermission(final UserManager userManager, final String imageTypeName,
-      final String userId,
-      final Type type) {
+  public boolean hasPermission(final String imageTypeName, final String userId, final Type type) {
```

Comment:
Can you run validatePermissionAndGetOwnerships inside of hasPermission method?
Most of the code here are redundant.

Reply:
They're not exactly the same. So hasPermission is based on authorizing image by Type, the type means imageCreate/imageGet/ImageDelete. But for ramp manager, it does not need detailed image operation level authorization. I tried to use Type.Admin but unfortunately, in the testing I didn't didn't find Admin is in User's Permission (even my ldap for az-platform-image) so we would get no imageOwnerships if we select ADMIN and for non azkaban admin it will fail.